omniscidb/ArrowStorage/ArrowStorage.cpp

```diff
@@ -852,10 +856,12 @@ void ArrowStorage::compareSchemas(std::shared_ptr<arrow::Schema> lhs,
     auto lhs_type = lhs_fields[i]->type();
     auto rhs_type = rhs_fields[i]->type();
 
-    if (!lhs_type->Equals(rhs_type)) {
-      throw std::runtime_error("Mismatched type for column "s + lhs_fields[i]->name() +
-                               ": "s + lhs_type->ToString() + " vs. "s +
-                               rhs_type->ToString());
+    if (!lhs_type->Equals(rhs_type) && !(lhs_type->id() == arrow::Type::NA) &&
```

Comment:
Why `!(a == b)` instead of `a != b`? Also, the NA type is not allowed for LHS.

Reply:
LHS not allowed to be NA, due to usage of `compareSchemas`. I think `compareSchemas` can be utility/static and should be symmetric to lhs/rhs.